index/scorch/snapshot_index.go

```diff
@@ -496,30 +496,55 @@ func docInternalToNumber(in index.IndexInternalID) (uint64, error) {
 
 func (i *IndexSnapshot) DocumentVisitFieldTerms(id index.IndexInternalID,
 	fields []string, visitor index.DocumentFieldTermVisitor) error {
+	_, err := i.documentVisitFieldTerms(id, fields, visitor, nil)
+	return err
+}
+
+func (i *IndexSnapshot) documentVisitFieldTerms(id index.IndexInternalID,
+	fields []string, visitor index.DocumentFieldTermVisitor, dvs segment.DocVisitState) (
+	segment.DocVisitState, error) {
 
 	docNum, err := docInternalToNumber(id)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	segmentIndex, localDocNum := i.segmentIndexAndLocalDocNumFromGlobal(docNum)
 	if segmentIndex >= len(i.segment) {
-		return nil
+		return nil, nil
 	}
 
 	ss := i.segment[segmentIndex]
 
 	if zaps, ok := ss.segment.(segment.DocumentFieldTermVisitable); ok {
+		if ss.dvState == nil {
+			ss.dvState = &fieldDocValueState{}
```

Comment:
the segment snapshot (ss) ought to immutable?  having a mutable dvState field here feels wrong?